lib/workers/branch/index.ts

```diff
@@ -362,115 +354,16 @@ export async function processBranch(
-                : cmd;
-
-              logger.debug({ cmd: compiledCmd }, 'Executing post-upgrade task');
-
-              const execResult = await exec(compiledCmd, {
-                cwd: config.localDir,
-              });
-
-              logger.debug(
-                { cmd: compiledCmd, ...execResult },
-                'Executed post-upgrade task'
-              );
-            } else {
-              logger.warn(
-                {
-                  cmd,
-                  allowedPostUpgradeCommands,
-                },
-                'Post-upgrade task did not match any on allowed list'
-              );
-            }
-          }
-
-          const status = await getRepoStatus();
-
-          for (const relativePath of status.modified.concat(status.not_added)) {
-            for (const pattern of fileFilters) {
-              if (minimatch(relativePath, pattern)) {
-                logger.debug(
-                  { file: relativePath, pattern },
-                  'Post-upgrade file saved'
-                );
-                const existingContent = await readLocalFile(relativePath);
-                const existingUpdatedArtifacts = config.updatedArtifacts.find(
-                  (ua) => ua.name === relativePath
-                );
-                if (existingUpdatedArtifacts) {
-                  existingUpdatedArtifacts.contents = existingContent;
-                } else {
-                  config.updatedArtifacts.push({
-                    name: relativePath,
-                    contents: existingContent,
-                  });
-                }
-                // If the file is deleted by a previous post-update command, remove the deletion from updatedArtifacts
-                config.updatedArtifacts = config.updatedArtifacts.filter(
-                  (ua) => ua.name !== '|delete|' || ua.contents !== relativePath
-                );
-              }
-            }
-          }
-
-          for (const relativePath of status.deleted || []) {
-            for (const pattern of fileFilters) {
-              if (minimatch(relativePath, pattern)) {
-                logger.debug(
-                  { file: relativePath, pattern },
-                  'Post-upgrade file removed'
-                );
-                config.updatedArtifacts.push({
-                  name: '|delete|',
-                  contents: relativePath,
-                });
-                // If the file is created or modified by a previous post-update command, remove the modification from updatedArtifacts
-                config.updatedArtifacts = config.updatedArtifacts.filter(
-                  (ua) => ua.name !== relativePath
-                );
-              }
-            }
-          }
-        }
-      }
+      config.updatedArtifacts = await postUpgradeCommandExecutor(
+        'update',
+        config
+      );
+      config.updatedArtifacts = await postUpgradeCommandExecutor(
+        'branch',
+        config
+      );
```

Comment:
Move this to simple if else in `postUpgradeCommandExecutor`. Why do we need two calls here?

Reply:
The looping logic inside `postUpgradeCommandExecutor` becomes a bit more complicated if it's moved inside.
But I'll move it next time I have time to work on this PR

Reply:
Okay actually wasn't too bad at all, pushed now should be good to go 👍